homeassistant/components/humidifier/const.py

```diff
@@ -0,0 +1,101 @@
+"""Provides the constants needed for component."""
+
+# All activity disabled / Device is off/standby
+OPERATION_MODE_OFF = "off"
+
+# The device supports humidifying/drying to a range
+OPERATION_MODE_HUMIDIFY_DRY = "humidify_dry"
+
+# The humidity is set based on a schedule, learned behavior, AI or some
+# other related mechanism. User is not able to adjust the target humidity
+OPERATION_MODE_AUTO = "auto"
+
+# Device is in Dry mode
+OPERATION_MODE_DRY = "dry"
+
+# Device is in Humidification/Misting mode
+OPERATION_MODE_HUMIDIFY = "humidify"
+
+# Only the fan is on, not fan and another mode like humidify
+OPERATION_MODE_FAN_ONLY = "fan_only"
+
+OPERATION_MODES = [
+    OPERATION_MODE_OFF,
+    OPERATION_MODE_HUMIDIFY_DRY,
+    OPERATION_MODE_AUTO,
+    OPERATION_MODE_DRY,
+    OPERATION_MODE_HUMIDIFY,
+    OPERATION_MODE_FAN_ONLY,
+]
+
+# No preset is active
+PRESET_NONE = "none"
+
+# Device is running an energy-saving mode
+PRESET_ECO = "eco"
+
+# Device is in away mode
+PRESET_AWAY = "away"
+
+# Device turn all valve full up
+PRESET_BOOST = "boost"
+
+# Device is in comfort mode
+PRESET_COMFORT = "comfort"
+
+# Device is in home mode
+PRESET_HOME = "home"
+
+# Device is prepared for sleep
+PRESET_SLEEP = "sleep"
+
+
+# Possible fan state
+FAN_ON = "on"
```

Comment:
This makes no sense, why can it be on, off, medium etc.  Where is the research that is done to get to this list ?

Reply:
I just decided it would be similar to HVACs so copied it from `climate`.
I will try to make a research...

Reply:
I will respond to this separately because the response is big.